Assembler.UnitTests/AssemblerTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Moq;
+using NUnit.Framework;
+
+namespace Assembler.UnitTests
+{
+    [TestFixture]
+    public class MessagesAssemblerTests
+    {
+        private Mock<IResolver<FrameType, IHandler<BaseFrame, BaseMessageInAssembly>>> _resolverMock;
+        private Mock<IConverter<BaseMessageInAssembly, BaseAssembledMessage>> _converterMock;
+        private Mock<ITimeBasedCache<BaseMessageInAssembly>> _cacheMock;
+        private Mock<IHandler<BaseFrame, BaseMessageInAssembly>> _firstHandlerMock;
+        private Mock<IHandler<BaseFrame, BaseMessageInAssembly>> _secondHandlerMock;
+        private Base.MessagesAssembler _assembler;
+
+        private List<BaseAssembledMessage> _assembledMessages;
+        private List<Mock<IHandler<BaseFrame, BaseMessageInAssembly>>> _handlersList;
+
+        [SetUp]
+        public void Setup()
+        {
+            _assembledMessages = new List<BaseAssembledMessage>();
+            _resolverMock = new Mock<IResolver<FrameType, IHandler<BaseFrame, BaseMessageInAssembly>>>();
+            _converterMock = new Mock<IConverter<BaseMessageInAssembly, BaseAssembledMessage>>();
+            _cacheMock = new Mock<ITimeBasedCache<BaseMessageInAssembly>>();
+
+            _firstHandlerMock = new Mock<IHandler<BaseFrame, BaseMessageInAssembly>>();
+            _secondHandlerMock = new Mock<IHandler<BaseFrame, BaseMessageInAssembly>>();
+
+            _handlersList = new List<Mock<IHandler<BaseFrame, BaseMessageInAssembly>>>
+            {
+                _firstHandlerMock,
+                _secondHandlerMock
+            };
+            var handlersList = _handlersList.Select(handlers => handlers.Object);
+
+            _assembler = new Base.MessagesAssembler(_resolverMock.Object, _cacheMock.Object, handlersList,
+                _converterMock.Object, Utilities.GetLoggerFactory());
+
+            _assembler.MessageAssembled += _assembledMessages.Add;
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            _resolverMock.VerifyNoOtherCalls();
+            _converterMock.VerifyNoOtherCalls();
+            _cacheMock.VerifyNoOtherCalls();
+            _firstHandlerMock.VerifyNoOtherCalls();
+            _secondHandlerMock.VerifyNoOtherCalls();
+        }
+
+        [Test]
+        public void Constructor_NoHandlers_NoExceptionBeingThrown()
```

Comment:
Rethinking about this - wouldn't it actually make more sense to throw something if no handlers were given? I mean, in that case, that either means:
1. The resolver has handlers, but these could never invoke something in the assembler, or
2. The resolver itself has no handlers, which in that case would mean that the assembler does nothing (or would throw an exception in the `Assemble(BaseFrame)` method)

I could imagine that maybe option 1 might be possible if you intentionally have handlers that do not invoke anything but other than that this situation should definitely raise some kind of alarm

Reply:
I think you are right, I throw an exception now

Reply:
👍 
Also cool one-liner in `SubscribeToEvents(...)`

Reply:
Nice, changed

Reply:
👍